sympy/core/function.py

```diff
@@ -1479,8 +1479,33 @@ def _subset(a, b):
 
             if _subset(old_vars, self_vars):
                 return Derivative(new, *(self_vars - old_vars).items())
-
-        return Derivative(*(x._subs(old, new) for x in self.args))
+        # Substitution remains unevaluate unless one of the following applies:
+        # 1) old is an instance of UndefinedFunction, rather than an expression
+        # 2) Neither old nor new term contains any self.variables
+        # 3) Substitution replaces one of self.variables by an expression whose
+        #    variable(s) are not present in self.args[0]
+        # 4) Substitution replaces one of self.variables by an expression that
+        #    does not depend on other variables
+        # 5) expr == old, complete replacement of differentiated expression
+        # 6) Substitution does not change either expr or variables
+        # See issue 13795
+
+        expr = self.args[0]
+        if isinstance(old, UndefinedFunction) or \
```

Comment:
I would use `_diff_wrt` here. `_diff_wrt` being defined on Function instead of UndefinedFunction is an issue that should be fixed separately from this.